agent/auto-config/auto_config.go

```diff
@@ -54,26 +54,20 @@ var (
 // then we will need to add some locking here. I am deferring that for now
 // to help ease the review of this already large PR.
 type AutoConfig struct {
-	builderOpts        config.BuilderOpts
+	acConfig           Config
 	logger             hclog.Logger
-	directRPC          DirectRPC
-	waiter             *lib.RetryWaiter
-	overrides          []config.Source
 	certMonitor        CertMonitor
 	config             *config.RuntimeConfig
 	autoConfigResponse *pbautoconf.AutoConfigResponse
 	autoConfigSource   config.Source
-	cancel             context.CancelFunc
 }
 
-// New creates a new AutoConfig object for providing automatic
-// Consul configuration.
-func New(config *Config) (*AutoConfig, error) {
-	if config == nil {
-		return nil, fmt.Errorf("must provide a config struct")
-	}
-
-	if config.DirectRPC == nil {
+// New creates a new AutoConfig object for providing automatic Consul configuration.
+func New(config Config) (*AutoConfig, error) {
+	switch {
```

Comment:
Just curious but why change the multiple conditionals to the switch with the same conditions on each branch?

Reply:
It's a very minor stylistic thing. I tend to use `switch` when the cases are mutually exclusive. Since each case returns it's not necessary, but the grouping of all the conditions into a single block may imply that only one block will ever evaluate.

I had to change the first one from `config` to `config.Loader`, so I change it to a switch at the same time.